apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/LoadedFilterManager.java

```diff
@@ -232,26 +233,25 @@ public synchronized void refreshFilters() {
             || this.firstBatchCreated.before(currentFirstBatchUpdate)) {
           LOGGER.info("Trimmed LoadedFile filters before {}", currentFirstBatchUpdate);
           List<LoadedFile> loadedFiles = fetchLoadedFiles();
-          this.filters = trimFilters(this.filters, loadedFiles);
-          this.firstBatchCreated = currentFirstBatchUpdate;
+          newFilters = trimFilters(newFilters, loadedFiles);
         }
 
-        // update the transaction time as well
-        this.transactionTime = currentLastBatchCreated;
+        set(newFilters, currentFirstBatchUpdate, currentFirstBatchUpdate);
```

Comment:
Now, instead of setting the filters and other states during the method, the state is set once at the end.

Reply:
Shouldn't the third parameter passed into `set` be `currentLastBatchCreated` rather than `currentFirstBatchUpdate` or am I misunderstanding the workflow?